site/docs/4.3/getting-started/javascript.md

```diff
@@ -139,3 +139,68 @@ Bootstrap's plugins don't fall back particularly gracefully when JavaScript is d
 All Bootstrap's JavaScript files depend on `util.js` and it has to be included alongside the other JavaScript files. If you're using the compiled (or minified) `bootstrap.js`, there is no need to include this—it's already there.
 
 `util.js` includes utility functions and a basic helper for `transitionEnd` events as well as a CSS transition emulator. It's used by the other plugins to check for CSS transition support and to catch hanging transitions.
+
+## Sanitizer
+
+Tooltips and Popovers use our built-in sanitizer to sanitize options which receive HTML.
+
+The default `whiteList` value is the following:
+
+{% highlight js %}
+var DefaultWhitelist = {
+  // Global attributes allowed on any supplied element below.
+  '*': ['class', 'dir', 'id', 'lang', 'role', ARIA_ATTRIBUTE_PATTERN],
+  a: ['target', 'href', 'title', 'rel'],
+  area: [],
+  b: [],
+  br: [],
+  col: [],
+  code: [],
+  div: [],
+  em: [],
+  hr: [],
+  h1: [],
+  h2: [],
+  h3: [],
+  h4: [],
+  h5: [],
+  h6: [],
+  i: [],
+  img: ['src', 'alt', 'title', 'width', 'height'],
+  li: [],
+  ol: [],
+  p: [],
+  pre: [],
+  s: [],
+  small: [],
+  span: [],
+  sub: [],
+  sup: [],
+  strong: [],
+  u: [],
+  ul: []
+}
+{% endhighlight %}
+
+If you want to add new values to this default whiteList you can do the following:
+
+{% highlight js %}
+var myDefaultWhiteList = $.fn.tooltip.Constructor.Default.whiteList
+
+// To allow table elements
+myDefaultWhiteList.table = []
+
+// To allow td elements and data-option attributes on td elements
+myDefaultWhiteList.td = ['data-option']
+{% endhighlight %}
+
+Maybe you want to disable our sanitizer because you prefer to use a dedicated library, for example <a href="https://www.npmjs.com/package/dompurify">DOMPurify</a>
+then you should do the following.
+
+{% highlight js %}
+$('#yourTooltip').tooltip({
+  sanitize: false,
+  template: DOMPurify.sanitize(yourDirtyTemplate),
```

Comment:
While I see why you're doing it this way, it makes it more difficult to actually implement custom (per tooltip) templates.

Before, one could simply do a global Data API implementation (e.g. `$('[data-toggle="tooltip"]').tooltip(...)`) and the template would be imported automatically as a data attribute. It's a rare case when tooltips are initialized individually.

With this current implementation, one would have to literally iterate over each individual tooltip just to extract the template and then sanitize it before initializing the tooltip.

This is highly inefficient, especially considering since the jQuery plugin already implements its own each loop:
```js
$('[data-toggle="tooltip"]').each(() => {
  var $el = $(this);
  $el.tooltip({
    sanitize: false,
    template: DOMPurify.sanitize($el.data('template'))
  })
```

An easier and more efficient approach is to simply support a function being passed to the `sanitize` option:
```js
$('[data-toggle="tooltip"]').tooltip({
  sanitize: DOMPurify.sanitize,
})
```

That way, no matter where the template is supplied, either passed as a JavaScript option upon initialization or via a data attribute, it will always go through the correct sanitization function.

Alternatively, if a sanitizer requires additional options or doesn't support the content being the first parameter, an anonymous function could be used:

```js
$('[data-toggle="tooltip"]').tooltip({
  sanitize: unsafeMarkup => someOtherXssFilter(sanitizeOptions, unsafeMarkup),
})
```

Reply:
I prefer to allow passing a function something like:

```js
$('[data-toggle="tooltip"]').tooltip({
  sanitizeFn: unsafeMarkup => someOtherXssFilter(sanitizeOptions, unsafeMarkup),
})
```
it's more flexible for our users

Reply:
So you're saying there would be three new options: `sanitize`, `sanitizeFn` and `whiteList`?

That's fine with me. I was just trying to keep it as simple as possible and follow the established option pattern (other options allow functions to be passed to them, no "Fn" suffix).

Reply:
yep exactly 👍 I pushed what I did